### PR TITLE
Search: Add missing JavaScript dependencies

### PR DESCRIPTION
**Pull request by jsnmoon**
Taken from #21637.

#### Changes proposed in this Pull Request:
* Changes the React dependency to version 16.
* Adds react-dom and react-redux dependencies.
* Adds and uses Gutenberg's browserslist as a dev dependency.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* No testing necessary.

### CODE REVIEW

**Review by kangzj — DISMISSED** [review]
Looks good 👍

**Review by jeherve — DISMISSED** [review]

**Review by dereksmart — APPROVED** [review]

**Comment by jsnmoon** [comment]
🚢 r236142-wpcom

File: projects/packages/search/package.json [hunk 1]
```diff
@@ -56,7 +56,9 @@
 		"preact": "10.5.15",
 		"q-flat": "1.0.7",
 		"qss": "2.0.3",
-		"react": "17.0.2",
+		"react": "16.14.0",
```
Comment (anomiex) [inline projects/packages/search/package.json:59]: Why are you downgrading React?

Comment (jsnmoon) [reply projects/packages/search/package.json:59]: Neither `@wordpress/block-editor` nor `@wordpress/components` accept React 17 as a valid peer dependency. I downgraded here to mitigate any risk of breaking changes at this time.

That said, we may need to revisit this change due to Jetpack components relying on React 17.

**Pull request merged**